document-generator-accounts/src/test/java/uk/gov/companieshouse/document/generator/accounts/AccountsDocumentInfoServiceImplTest.java

```diff
@@ -30,16 +32,13 @@
     @Mock
     private TransactionService transactionService;
 
+    @Mock
+    private AccountsService accountsService;
+
     @Test
     @DisplayName("Tests the unsuccessful retrieval of an accounts document data due to no accounts resource in transaction")
     void testUnsuccessfulGetDocumentInfoNoAccountsResourceInTransaction() {
-        Map<String, Resource> resources = new HashMap<>();
-        resources.put("error", createResource());
-
-        Transaction transaction = new Transaction();
-        transaction.setResources(resources);
-        when(transactionService.getTransaction(anyString())).thenReturn(transaction);
-
+        when(transactionService.getTransaction(anyString())).thenReturn(createTransaction());
         assertNull(accountsDocumentInfoService.getDocumentInfo());
```

Comment:
Should there also be an accountService call here too? Which, given the context of the test, would return null I would assume?

Reply:
Nope, so that is why it is empty. The ones that has an accounts resource are those that have an accounts resource valid link. The resource that is empty is not a valid resource.